src/components/application_manager/src/commands/mobile/show_request.cc

```diff
@@ -125,38 +161,54 @@ void ShowRequest::Run() {
       smart_objects::SmartObject(smart_objects::SmartType_Array);
 
   int32_t index = 0;
+  int32_t main_field_1_index = -1;
   if ((*message_)[strings::msg_params].keyExists(strings::main_field_1)) {
     msg_params[hmi_request::show_strings][index][hmi_request::field_name] =
         static_cast<int32_t>(hmi_apis::Common_TextFieldName::mainField1);
     msg_params[hmi_request::show_strings][index][hmi_request::field_text] =
         (*message_)[strings::msg_params][strings::main_field_1];
+    main_field_1_index = index;
```

Comment:
Well, that's unfortunate. Would like to have a cleaner approach to this, but I don't really see one at the moment